test/requests.js

```diff
@@ -1200,6 +1200,73 @@ const tests = function(web3) {
 
       assert.strictEqual(result.result.input, "0x");
     });
+
+    it("should return null for blockHash, blockNumber, and transactionIndex on a pending tx", async function() {
+      const send = pify(web3._provider.send.bind(web3._provider));
+
+      const txData = {
+        to: accounts[0],
+        from: accounts[1]
+      };
+
+      await send({
+        id: new Date().getTime(),
+        jsonrpc: "2.0",
+        method: "miner_stop"
+      });
+
+      let result = await send({
+        id: new Date().getTime(),
+        jsonrpc: "2.0",
+        method: "eth_sendTransaction",
+        params: [txData]
+      });
+
+      let transactionObject = await send({
+        id: new Date().getTime(),
+        jsonrpc: "2.0",
+        method: "eth_getTransactionByHash",
+        params: [result.result]
+      });
+
+      assert.strictEqual(null, transactionObject.result.blockHash);
+      assert.strictEqual(null, transactionObject.result.blockNumber);
+      assert.strictEqual(null, transactionObject.result.transactionIndex);
+    });
+
+    it("should return values for blockHash, blockNumber, and transactionIndex once mined", async function() {
```

Comment:
Looks like this test needs to be updated so it does mine the block, then does `eth_getTransactionByHash`, then checks that the values are present. Actually, you can go ahead and combine what this test here intended to test with the previous test; just add a call to `evm_mine` after the `asserts` then get call `eth_getTransactionByHash` again and check that it has the values you expect.